From 7322c8f478e4057b96c53cf4d645fc4079c27d56 Mon Sep 17 00:00:00 2001
From: Eliware <eli-sterling@users.noreply.github.com>
Date: Sat, 15 Aug 2026 00:00:00 +0000
Subject: [PATCH] build: detect Python 3.12 correctly

Python 3.12 requires four characters from sys.version to preserve the full
major/minor version. The old [:3] expression turns 3.12 into 3.1.

Upstream commit 448c19095c removed contrib/aclocal/python.m4 and switched to
the host Automake/Autoconf macros. The packaging builders skip this legacy
patch when that file is absent or already contains the corrected expression.
---
 contrib/aclocal/python.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contrib/aclocal/python.m4 b/contrib/aclocal/python.m4
index a39a900904..0df654cb6c 100644
--- a/contrib/aclocal/python.m4
+++ b/contrib/aclocal/python.m4
@@ -85,7 +85,7 @@ python2.1 python2.0])
   dnl library.

   AC_CACHE_CHECK([for $am_display_PYTHON version], [am_cv_python_version],
-    [am_cv_python_version=`$PYTHON -c "import sys; sys.stdout.write(sys.version[[:3]])"`])
+    [am_cv_python_version=`$PYTHON -c "import sys; sys.stdout.write(sys.version[[:4]])"`])
   AC_SUBST([PYTHON_VERSION], [$am_cv_python_version])

   dnl Use the values of $prefix and $exec_prefix for the corresponding
--
2.43.0
